move/open_art_market/sources/open_art_market.move

```diff
@@ -196,7 +202,46 @@ module open_art_market::open_art_market {
         
     }
 
-    // @todo: do we need a split method?
+    // Function to split a shard into two shards
+    public fun split_artwork_shard(artwork: &mut Artwork, artwork_shard: &mut ArtworkShard, shares: u64, ctx: &mut TxContext) {
+        let sender = tx_context::sender(ctx);
+
+        // Ensure that the shard belongs to the artwork
+        assert!(object::uid_to_inner(&artwork.id) == artwork_shard.artwork_id, EInvalidArtwork);
+
+        // Ensure that the sender is a shareholder of the artwork
+        assert!(df::exists_(&artwork.id, sender) == true, ECallerNotAShareHolder);
+
+        // Ensure that the original shard has sufficient shares to split
+        assert!(artwork_shard.shares > shares, EInsufficientShares);
+        
+        let shares2 = artwork_shard.shares - shares;
+
+        // Create new shard with remaining shares
+        let artwork_shard2 = ArtworkShard {
+            id: object::new(ctx),
+            artwork_id: artwork_shard.artwork_id,
+            shares: shares2,
+            incoming_price: (artwork_shard.incoming_price / artwork_shard.shares) * shares2,
+            artwork_name: artwork_shard.artwork_name,
+            artist: artwork_shard.artist,
+            creation_date: artwork_shard.creation_date,
+            description: artwork_shard.description,
+            image_url: artwork_shard.image_url,
+        };
+
+        // Update shares and incoming price of original shard
+        artwork_shard.incoming_price = (artwork_shard.incoming_price / artwork_shard.shares) * shares;
+        artwork_shard.shares = shares;
+
+        // Transfer new shard to sender
+        transfer::transfer(artwork_shard2, tx_context::sender(ctx));
```

Comment:
I think its more intuitive that the created share object is the one with the shares requested. The result is the same but the new object should contain the shares that were requested (similar to how we split coins)